tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -364,6 +368,168 @@ public void testRsaWithoutCrtParams() throws Exception {
+        assertEquals("SHA-1", spec.getDigestAlgorithm());
+        assertEquals("SHA-1", ((MGF1ParameterSpec) spec.getMGFParameters()).getDigestAlgorithm());
+
+        // setting null params OK, shouldn't change from default
+        signature = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        signature.setParameter(null);
+        signature.initSign(pair.getPrivate());
+        spec = signature.getParameters().getParameterSpec(PSSParameterSpec.class);
+        assertEquals("SHA-1", spec.getDigestAlgorithm());
+        assertEquals("SHA-1", ((MGF1ParameterSpec) spec.getMGFParameters()).getDigestAlgorithm());
+    }
+
+    @Test
+    void testRsaPSSTryUpdateParamDuringBuffer() throws Exception {
+        final KeyPairGenerator kg = KeyPairGenerator.getInstance("RSA", AmazonCorrettoCryptoProvider.INSTANCE);
+        kg.initialize(2048);
+        final KeyPair pair = kg.generateKeyPair();
+        final Signature signer = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        final Signature verifier = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        final PSSParameterSpec spec1 = new PSSParameterSpec("SHA-224", "MGF1", MGF1ParameterSpec.SHA224, 224/8, 1);
+        final PSSParameterSpec spec2 = new PSSParameterSpec("SHA-256", "MGF1", MGF1ParameterSpec.SHA256, 256/8, 1);
+
+        // Initialize signatures and set non-default PSS parameters
+        signer.initSign(pair.getPrivate());
+        verifier.initVerify(pair.getPublic());
+        signer.setParameter(spec1);
+        verifier.setParameter(spec1);
+
+        // Assert that set values persisted
+        assertEquals(
+            spec1.getDigestAlgorithm(),
+            signer.getParameters().getParameterSpec(PSSParameterSpec.class).getDigestAlgorithm()
+        );
+        assertEquals(
+            spec1.getDigestAlgorithm(),
+            verifier.getParameters().getParameterSpec(PSSParameterSpec.class).getDigestAlgorithm()
+        );
+        assertEquals(
+            ((MGF1ParameterSpec) spec1.getMGFParameters()).getDigestAlgorithm(),
+            ((MGF1ParameterSpec) signer.getParameters().getParameterSpec(PSSParameterSpec.class).getMGFParameters()).getDigestAlgorithm()
+        );
+        assertEquals(
+            ((MGF1ParameterSpec) spec1.getMGFParameters()).getDigestAlgorithm(),
+            ((MGF1ParameterSpec) verifier.getParameters().getParameterSpec(PSSParameterSpec.class).getMGFParameters()).getDigestAlgorithm()
+        );
+
+        // Update message digest, assert that we can't upate params, and that signature verifies.
+        signer.update(MESSAGE);
+        verifier.update(MESSAGE);
+        assertThrows(IllegalStateException.class, () -> signer.setParameter(spec2));
+        assertThrows(IllegalStateException.class, () -> verifier.setParameter(spec2));
+        assertTrue(verifier.verify(signer.sign()));
+
+        // Finally, assert that we can set params after reinit (verify doesn't need explicit re-init)
```

Comment:
You shouldn't need to explicitly re-init signer either (as calling `.sign()` implicitly re-inits it).

Reply:
reinit removed.